### ISSUE TITLE
Consider providing a dynamic decoder

### ISSUE THREAD

**Issue by fwgreen** [jrstrunk/tempo#1]
Support for [gleam/dynamic](https://hexdocs.pm/gleam_stdlib/gleam/dynamic.html) might simplify working with libraries like [pgo](https://github.com/lpil/pgo) for instance.

Thanks for all your hard work!

**Comment by jrstrunk** [comment]
Good idea! I haven't used date fields in postgresql with gleam myself, would it decode an int (timestamp) or string (iso 8601) to get the date? Or just try both?

**Comment by fwgreen** [comment]
String might be more flexible as PGO currently doesn't support timestamps with a timezone.

**Comment by jrstrunk** [comment]
Just released v2 with this feature for the datetime and date modules! Let me know what you think. Since timestamps can have different levels of precision, I had to make a different function for each datetime representation (string, unix timestamp, unix timestamp milli, unix timestamp micro) instead of just having one function do it all. I also realized some errors were redundant and removed them, bumping the package to v2 (you will have to explicitly upgrade to v2 to get these changes).
